python/paddle/distributed/checkpoint/metadata.py

```diff
@@ -24,6 +24,7 @@ class LocalTensorMetadata:
 
     global_offset: Tuple[int]
     local_shape: Tuple[int]
+    dtype: str
```

Comment:
why we need dtype?

Reply:
In static mode, the ``state_dict`` is incomplete, so the previous code would trigger  ``key error``. Here, the ``dtype`` is stored in advance.